devitofwi/waveengine/acoustic.py

```diff
@@ -283,7 +301,12 @@ def _mod_oneshot(self, isrc: int, dt: float = None) -> NDArray:
         # Update source location in geometry
         geometry = self.geometry1shot
         geometry.src_positions[0, :] = self.geometry.src_positions[isrc, :]
-
+        if self.streamer_acquisition:
```

Comment:
I am a bit unsure about this. Is it correct to say that the `rec_x` and `rec_z` provided to the class during initalization are somehow ignored, we only 'learn' from the about the spacing `dr` and number of receivers `nr`, but the actual receiver grid is created here from the source with a step of dr?

In practice, one always has a longer source-to-first-receiver offset than the distance between receivers... what about using rec_x and rec_z to provide a relative location wrt to source and then create here the axis based on them, e.g.
```
dr = 20
nr = 100
s_to_r = 30 # first offset
rec_x = np.arange(nr) * dr + s_to_r
...
amod = AcousticWave2D(..., rec_x, ...)
```
and here
```
if self.streamer_acquisition:
       # Update receiver locations in geometry
       geometry.rec_positions[:, 0] = geometry.src_positions[0, 0] + geometry.rec_positions[:, 0]
```

It would also be good to mention this in the docstring of the class as right now the comment on `streamer_acquisition` does not give away how the actual receiver grid is built.

Reply:
Yes, I agree that the code is not flexible at the moment. So, I think using the position of the source and the geometry of the first shot, as you showed in the code below, will suffice. If this is the case, I will go ahead and make the changes?
```
if self.streamer_acquisition:
       # Update receiver locations in geometry
       geometry.rec_positions[:, 0] = geometry.src_positions[0, 0] + geometry.rec_positions[:, 0]
```

Reply:
Sure, go ahead :) I think in most cases the relative geometry will always be the same, so we can start this way… in future we may consider to support also a case where the geometry changes from shot to shot to account for real positions of receivers, but let’s leave it out for now